src/Files.App/Helpers/QuickLookHelpers.cs

```diff
@@ -1,35 +1,68 @@
-using Files.Shared.Extensions;
-using System.Diagnostics;
+using System;
+using System.IO;
+using System.IO.Pipes;
+using System.Security.Principal;
 using System.Threading.Tasks;
-using Windows.Foundation.Collections;
 
-namespace Files.App.Helpers
+namespace Files.App.Helpers;
+
+public static class QuickLookHelpers
 {
-    public static class QuickLookHelpers
+    public static Task ToggleQuickLook(IShellPage associatedInstance, bool switchPreview = false)
     {
-        public static async Task ToggleQuickLook(IShellPage associatedInstance, bool switchPreview = false)
+        if (!App.AppModel.IsQuickLookSupported || !associatedInstance.SlimContentPage.IsItemSelected || associatedInstance.SlimContentPage.IsRenamingItem)
+            return Task.CompletedTask;
+
+        App.Logger.Info("Toggle QuickLook");
+
+        string PipeName = $"QuickLook.App.Pipe.{WindowsIdentity.GetCurrent().User?.Value}";
+        string Message = switchPreview ? "QuickLook.App.PipeMessages.Switch" : "QuickLook.App.PipeMessages.Toggle";
+
+        using var client = new NamedPipeClientStream(".", PipeName, PipeDirection.Out);
+        client.Connect();
```

Comment:
To be checked what happens if you close quicklook after starting Files. We might want to change this to `client.Connect(500);`.